### ISSUE TITLE
Simplifying the documentation

### ISSUE THREAD

**Issue by karlnapf** [shogun-toolbox/shogun#2602]
We currently have documentation cluttered around 3 places
- Readmes
- Doxygen
- Wiki

There are lots of redundant parts in those three, and lots of things are outdated. I suggest we clean this up a little. Main point would be to get rid of redundant parts, structure the remaining things a bit more.
Here are some thoughts:
- Doxygen is the best place for documenting code, i.e. [class interfaces](http://www.shogun-toolbox.org/doc/en/latest/namespaceshogun.html), [documented examples](#2600), and everything that involves code (maybe even developer guidelines/tutorial. The cool thing here is that we can put the code in examples and generate doxygen files from this, so things are kept up to date automatically.
  In my eyes, there should be nothing else in here because it is easier to maintain things elsewise (see below). Tutorials about using Shogun, such as the [OpenCV tutorial](#2489) should go into doxygen as well as we can then make sure the code actually runs.
- The wiki is a good place to quickly change texts (which is harder for doxygen). Therefore, it should contain all "soft" documents about Shogun, such as a project description, contributors, overview texts, feature lists etc. There are currently lots of such things in the doxygen -- and all of them are outdated. Via putting them into the Shogun wiki, we make it much easier to maintain such files.
- Readme files are already in synch with the wiki, so they are a subset of it. The `doc/md` folder is a git submodule which ensures this. This is cool as we then still have local readme files while at the same time it is easy to edit them via the github wiki and we get pretty rendering via markdown.

We can put links that relate doxygen and wiki/md to allow for easy going back and forth. It is also possible to render our markdown readme files in the wiki, which is for example done for the quickstart guide [here](http://www.shogun-toolbox.org/doc/en/latest/quickstart.html). 

Benefits: Only having to maintain some Readme files, which is possible over the wiki. Everything else is automatically generated from there. In addition, all documented code can be executed and therefore keeps on working

**Comment by abinashmeher999** [comment]
@karlnapf Has everything been done here?

**Comment by karlnapf** [comment]
@abinashmeher999 Not at all! Check the things I listed above and the example patch I referenced
Sorry for late reply
